Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/PrefabSystemComponent.cpp

```diff
@@ -631,13 +631,16 @@ namespace AzToolsFramework
                 //member itself, so we need to move instancesValue to the correct position for the next insert
                 memberFound = instancesValue->get().FindMember(PrefabDomUtils::InstancesName);
                 instancesValue = memberFound->value;
-                instancesValue->get().SetObject();
             }
             else
             {
                 instancesValue = memberFound->value;
             }
 
+            if (!instancesValue->get().IsObject())
+            {
+                instancesValue->get().SetObject();
```

Comment:
I couldn't find a reason for adding this change in the description of the PR. Can you explain why this fix was needed now? Are the sanitation changes requiring this fix?

Reply:
A defaulted empty Instances container in json is of type array. The below operations would fail if we did not change the type. Since default values can now appear in our Doms we need to type check our containers.